import-scripts/filter_non_somatic_events_py3.py

```diff
@@ -0,0 +1,136 @@
+"""
+
+import argparse
+from enum import Enum
+from generate_az_study_changelog_py3 import DataHandler
+
+
+class EventType(Enum):
+    """An Enum class to represent mutation or structural variant event types."""
+
+    MUTATION = 1
+    STRUCTURAL_VARIANT = 2
+
+
+class LineProcessor:
+    """Handles the processing of each line - filtering for somatic events only"""
+
+    def __init__(self, event_type, col_indices, output_file_handle):
+        self.event_type = event_type
+        self.col_indices = col_indices
+        self.output_file_handle = output_file_handle
+        self.header_was_written = False
+
+    def line_is_commented(self, line):
+        """Determines if the given line in the file is a comment.
+
+        Args:
+            line (string): A line from the input file
+
+        Returns:
+            boolean: True or False, indicating whether the line is a comment
+        """
+        return line[0] == '#'
+
+    def process(self, line):
+        """Process each line of the given file to remove all events that are not 'SOMATIC'.
+
+        Args:
+            line (string): A line from the input file
+
+        Raises:
+            IndexError: If 'Mutation_Status' column is not found in mutation file
+            IndexError: If 'SV_Status' column is not found in structural variant file
+        """
+        if self.line_is_commented(line):
+            self.output_file_handle.write(line)
+            return
+
+        if not self.header_was_written:
+            self.output_file_handle.write(line)
+            self.header_was_written = True
+            return
+
+        col_index = None
+        if event_type == EventType.MUTATION:
+            if 'Mutation_Status' not in self.col_indices:
+                raise IndexError('Unable to find Mutation_status column in event file')
+            col_index = self.col_indices['Mutation_Status']
+        elif event_type == EventType.STRUCTURAL_VARIANT:
+            if 'SV_Status' not in self.col_indices:
+                raise IndexError('Unable to find SV_Status column in event file')
+            col_index = self.col_indices['SV_Status']
+
+        cols = line.split('\t')
+        value = cols[col_index].rstrip('\n')
+        if value.casefold() == 'SOMATIC'.casefold():
+            self.output_file_handle.write(line)
+
+
+class FilteredFileWriter:
+    """Handles writing the filtered file containing only somatic events"""
+
+    def __init__(self, input_file_path, output_file_path, event_type):
+        self.input_file_path = input_file_path
+        self.output_file_path = output_file_path
+        self.event_type = event_type
+        self.data_handler = DataHandler(input_file_path)
+        self.col_indices = self.data_handler.get_col_indices({"Mutation_Status", "SV_Status"})
+
+    def write(self):
+        """Processes the input file and writes out a filtered version including only somatic events"""
+        input_file_handle = open(input_file_path, "r")
```

Comment:
open file using the with command .. and avoid the close statements below. All logic can be in the with block. With two file handles, nest the with blocks.